plugins/inputs/azure_monitor/collector.go

```diff
@@ -0,0 +1,386 @@
+	}
+
+	return metricNameLocalizedValue, nil
+}
+
+func getMetricsClientMetricUnit(metric *armmonitor.Metric) (*string, error) {
+	if metric == nil {
+		return nil, fmt.Errorf("metrics client response is bad formatted: metric is missing")
+	}
+
+	if metric.Unit == nil {
+		return nil, fmt.Errorf("metrics client response is bad formatted: metric Unit is missing")
+	}
+
+	metricUnit := string(*metric.Unit)
+	return &metricUnit, nil
+}
+
+func getMetricsClientResponseNamespace(response *armmonitor.MetricsClientListResponse) (*string, error) {
+	if response.Namespace == nil {
+		return nil, fmt.Errorf("metrics client response is bad formatted: reponse Namespace is missing")
+	}
+
+	return response.Namespace, nil
+}
+
+func getMetricsClientResponseResourceRegion(response *armmonitor.MetricsClientListResponse) (*string, error) {
+	if response.Resourceregion == nil {
+		return nil, fmt.Errorf("metrics client response is bad formatted: reponse Resourceregion is missing")
+	}
+
+	return response.Resourceregion, nil
+}
+
+func getMetricsClientMetricValueFields(metricValue *armmonitor.MetricValue) map[string]interface{} {
+	if metricValue == nil {
+		return nil
+	}
+
+	if metricValue.TimeStamp == nil {
+		return nil
+	}
+
+	metricFields := make(map[string]interface{})
+	metricValueFieldsNum := 1
+
+	metricFields["timeStamp"] = metricValue.TimeStamp.Format("2006-01-02T15:04:05Z07:00")
+
+	if metricValue.Total != nil {
+		metricFields["total"] = *metricValue.Total
+		metricValueFieldsNum++
+	}
+
+	if metricValue.Average != nil {
+		metricFields["average"] = *metricValue.Average
+		metricValueFieldsNum++
+	}
+
+	if metricValue.Count != nil {
+		metricFields["count"] = *metricValue.Count
+		metricValueFieldsNum++
+	}
+
+	if metricValue.Minimum != nil {
+		metricFields["minimum"] = *metricValue.Minimum
+		metricValueFieldsNum++
+	}
+
+	if metricValue.Maximum != nil {
+		metricFields["maximum"] = *metricValue.Maximum
+		metricValueFieldsNum++
+	}
+
+	if metricValueFieldsNum < minMetricsFields {
+		return nil
+	}
+
+	return metricFields
+}
+
+func createMetricName(metric *armmonitor.Metric, response *armmonitor.MetricsClientListResponse) (*string, error) {
+	namespace, err := getMetricsClientResponseNamespace(response)
+	if err != nil {
+		return nil, err
+	}
+
+	name, err := getMetricsClientMetricNameLocalizedValue(metric)
+	if err != nil {
+		return nil, err
+	}
+
+	replacer := strings.NewReplacer(".", "_", "/", "_", " ", "_", "(", "_", ")", "_")
+	metricName := fmt.Sprintf("azure_monitor_%s_%s",
+		replacer.Replace(strings.ToLower(*namespace)),
+		replacer.Replace(strings.ToLower(*name)))
+
+	return &metricName, nil
+}
+
+func getMetricFields(metricValues []*armmonitor.MetricValue) map[string]interface{} {
+	for index := len(metricValues) - 1; index >= 0; index-- {
+		metricFields := getMetricsClientMetricValueFields(metricValues[index])
+		if metricFields == nil {
+			continue
+		}
+
+		return metricFields
+	}
```

Comment:
Is there a reason for this type of iteration? Why do you return the "latest" valid value? A comment would help the reader here.